examples/pytorch/preprocessing_dist_training/metis_creation.py

```diff
@@ -0,0 +1,89 @@
+import pandas as pd
+import os
+import glob
+import json
+import argparse
+from collections import defaultdict
+
+path = os.getcwd()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-n", "--name", help="name of graph to create", default="order")
+parser.add_argument("-nc", "--node_column", nargs="+", default=['order_id', 'entity_index', 'order_datetime', 'cid'])
+parser.add_argument("-nk", "--node_key", default='entity_index')
+parser.add_argument("-ec", "--edge_column", nargs="+", default=['predicate_type', 'predicate_index', 'entity_index', 'entity_index_y'])
+parser.add_argument("-es", "--edge_start", default="entity_index")
+parser.add_argument("-en", "--edge_end", default="entity_index_y")
+args = parser.parse_args()
+
+
+#Store all types of node in nodes folder
+nodes_list = sorted(glob.glob(os.path.join(path, "nodes/*")))
+
+if os.path.exists("{}_nodes.txt".format(args.name)):
+    os.remove("{}_nodes.txt".format(args.name))
+
+schema_dict = defaultdict(dict)
+
+node_type_id = 0
+all_nodes_count = 0
+for node_type_name in nodes_list:
+    nodes_count = 0
+    csv_files = sorted(glob.glob(os.path.join(node_type_name, "*.csv")))
+    for file_name in csv_files:
+        df = pd.read_csv(file_name, error_bad_lines=False, escapechar='\\', names=args.node_column, usecols=[*range(len(args.node_column))])
+        df_entity = pd.DataFrame(df[args.node_key], columns=[args.node_key])
+        df_entity['type'] = node_type_id
+        column_list = ['type']
+        for weight_index in range(len(nodes_list)):
+            weight_num = "weight{}".format(weight_index)
+            column_list.append(weight_num)
+            if weight_index == node_type_id:
+                df_entity[weight_num] = 1
+            else:
+                df_entity[weight_num] = 0
+        nodes_count += len(df_entity.index)
+        column_list.append(args.node_key)
+        df_entity.to_csv("{}_nodes.txt".format(args.name), columns=column_list, sep=" ", index=False, header=False, mode='a')
```

Comment:
why do you save it back? i'm confused what this for loop tries to do. could you add some comments?

Reply:
added comment for the same.